src/libs/actions/IOU.js

```diff
@@ -291,7 +312,12 @@ function payIOUReport({
             }
             Onyx.merge(ONYXKEYS.IOU, {error: true});
         })
-        .finally(() => Onyx.merge(ONYXKEYS.IOU, {loading: false})),
+        .finally(() => {
+            Onyx.merge(ONYXKEYS.IOU, {loading: false});
+            if (isSendingMoney) {
+                Navigation.navigate(ROUTES.REPORT);
```

Comment:
This comment could be improved. For the `Send Money` flow we wanted to mirror how the `Request` flow works, which [returns the user to their chat ](https://github.com/Expensify/App/blob/6433741e0080f9767602829871c1ea34d4ec6de3/src/libs/actions/IOU.js#L87)after clicking "Request".

Reply:
The confirmation page is still the existing one and if we don't return the user to the chat then they would just stay on that page with the green `Pay With Expensify` button available to be clicked on again.

Reply:
Ok, that makes more sense. But kind of another one of those sort of weird cases where we are combining the behaviors of two flows into one method and a clear line of sight about what makes those two flows different is hard to find e.g. if we moved this redirect to the call here instead...

https://github.com/Expensify/App/blob/7cfa7aba268c20a28d692f8ae999edd31491d7dd/src/pages/iou/IOUModal.js#L288-L298

Then it would be much easier to make the connection and we wouldn't even need to mention that 

>Usually if we are just creating an IOU, we stay on the confirmation page

Which might not always be true if someone later changes the behavior but does not update the comment.

Reply:
> where we are combining the behaviors of two flows into one method

> weird cases where we are combining the behaviors of two flows into one method
I vaguely remember wanting two separate flows, but in design doc process we "simplified" to combining flows. Now actually implementing I definitely see the hit it takes on readability.

Maybe this was what you were suggesting earlier and I misinterpreted, but I should create a new method in `src/libs/actions/IOU` called `IOU.sendMoney` that calls `payIOUReport` then redirects. 

We wouldn't want to have the redirect be directly in IOUModal.js because that breaks app philosophy, or can I just put redirect there?

Reply:
>We wouldn't want to have the redirect be directly in IOUModal.js because that breaks app philosophy, or can I just put redirect there?

Yeah, basically! I think the latest on that is we don't want to call `.then()` on any actions inside views, but it's fine to do this in another action.

Reply:
sweet, thanks Marc